### PR TITLE
Entity3371

### PR DESCRIPTION
**Pull request by saravanpa-aot**
*Issue #:*

https://github.com/bcgov/entity/issues/1311
https://github.com/bcgov/entity/issues/515
https://github.com/bcgov/entity/issues/3371

*Description of changes:*

1) separated out components for basic and premium components
2) a basic address component , bcol login component and selector component is added
3) handled the redirections

Co-authored-by: @nitheesh-aot 

*Checklist:*
I confirm that below checklist items are addressed in this pull request; (check the boxes applicable)
- [ ] No lint errors
- [ ] No test case failures and proper coverage for all modules/classes
- [ ] Updated the deployment configs for new environment variable(s)
- [ ] Updtaed the postman collection in entity repository (https://github.com/bcgov/entity/tree/master/api-e2e/postman) for e2e tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/components/auth/CreateAccount/AccountTypeSelector.vue [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+@Component({
+  components: {
+
+  },
+  methods: {
+    ...mapMutations('org', ['setSelectedAccountType', 'setCurrentOrganization'])
+  }
+})
+export default class AccountTypeSelector extends Vue {
+  private readonly ACCOUNT_TYPE = {
```
Comment (saravanpa-aot) [inline auth-web/src/components/auth/CreateAccount/AccountTypeSelector.vue:91]: @nitheesh-aot can you read it from the Enum?

**Comment by saravanpa-aot** [comment]
> Please do squash commits as there are 27 commits for this change. Also look at the sonar code smells.

looking at sonar now

File: auth-web/src/components/auth/BcolLogin.vue [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  }
+  private usernameRules = [
+    v => !!v.trim() || 'Username is required'
+  ]
+
+  private passwordRules = [
+    value => !!value || 'Password is required'
+  ]
+
+  mounted () {
```
Comment (sutherlanda) [inline auth-web/src/components/auth/BcolLogin.vue:91]: Shouldn't need to do this - you have initializer values.

File: auth-web/src/components/auth/CreateAccount/AccountCreateBasic.vue [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+  private errorMessage: string = ''
+  private saving = false
+  private readonly createOrg!: (
+    requestBody: CreateRequestBody
+  ) => Promise<Organization>
+  private readonly syncMembership!: (orgId: number) => Promise<Member>
+  private readonly syncOrganization!: (orgId: number) => Promise<Organization>
+  private readonly currentOrganization!: Organization
+  @Prop() stepForward!: () => void
+  @Prop() stepBack!: () => void
```
Comment (sutherlanda) [inline auth-web/src/components/auth/CreateAccount/AccountCreateBasic.vue:89]: I have a PR pending that once merged, you will be able to just extend a mixin `Steppable` that will remove the need to manually specific these.

File: auth-web/src/components/auth/CreateAccount/AccountCreatePremium.vue [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+  private readonly currentOrganization!: Organization
+  private readonly currentUser!: KCUserProfile
+  @Prop() stepForward!: () => void
+  @Prop() stepBack!: () => void
+  private readonly setCurrentOrganization!: (organization: Organization) => void
+  private readonly setCurrentOrganizationAddress!: (address: Address) => void
+  private readonly setGrantAccess!: (grantAccess: boolean) => void
+
+  async mounted () {
+    // this.setCurrentOrganization(undefined)
```
Comment (sutherlanda) [inline auth-web/src/components/auth/CreateAccount/AccountCreatePremium.vue:152]: Stray comment, can remove the whole hook method.

File: auth-web/src/components/auth/CreateAccount/AccountCreateBasic.vue [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+  private readonly teamNameRules = [v => !!v || 'An account name is required']
+  private isFormValid (): boolean {
+    return !!this.teamName
+  }
+  private async save () {
+    // Validate form, and then create an team with this user a member
+    if (this.isFormValid()) {
+      const createRequestBody: CreateRequestBody = {
+        name: this.teamName,
+        typeCode: this.teamType === 'BASIC' ? 'IMPLICIT' : 'EXPLICIT'
```
Comment (sutherlanda) [inline auth-web/src/components/auth/CreateAccount/AccountCreateBasic.vue:102]: Do we have an enum for account type? If not, we should make one

File: auth-web/src/store/modules/org.ts [hunk 6]
```diff
@@ -120,6 +140,12 @@ export default class OrgModule extends VuexModule {
     return response?.data
   }
 
+  @Action({ rawError: true })
+  public async validateBcolAccount (bcolProfile: BcolProfile): Promise<BcolAccountDetails> {
+    const response = await BcolService.validateBCOL(bcolProfile)
```
Comment (sutherlanda) [inline auth-web/src/store/modules/org.ts:145]: Just return the response, no need to store in const.

File: auth-web/src/util/constants.ts [hunk 7]
```diff
@@ -41,7 +41,9 @@ export enum Pages {
 }
 
 export enum Account {
-    ANONYMOUS = 'ANONYMOUS'
+    ANONYMOUS = 'ANONYMOUS',
```
Comment (sutherlanda) [inline auth-web/src/util/constants.ts:44]: Use this enum where you can.  Extend it with 'IMPLICIT' / 'EXPLICIT' until we remove those.

**Pull request merged**